### DISCUSSION TITLE
可以像用vue写网页一样，编写vue组件，并且直接用到md文档中吗？

### DISCUSSION THREAD

**Discussion by Chengyunlai** [vuepress/core#1019] [Q&A] [answered]
[Markdown 与 Vue SFC](https://v2.vuepress.vuejs.org/zh/advanced/cookbook/markdown-and-vue-sfc.html)
这样子的写法 好像有点困难

**Comment by Mister-Hope** [comment]
困难在哪？

**Reply by Chengyunlai** [reply]
因为看到了`<Badge text="演示" />`这个演示，尝试了一下写一个`component.vue`，但不能直接在Md文档中用，是因为没配置吗

**Reply by Mister-Hope** [reply]
您导入或者全局注册了么？

**Reply by Mister-Hope** [reply]
还是您想跟我说你用原生的 vue 框架，随便写一个组件扔在项目里不注册不导入就可以用？

# ?

**Comment by Chengyunlai** [comment, marked as answer]
嗯..感谢你的耐心回答:joy:
我再翻了翻手册看到下面的步骤，然后成功了..其实还是我文档没看仔细。问题解决了
- **STEP1**

![image](https://user-images.githubusercontent.com/58059527/183351476-ce292e3c-a5e0-4c04-9803-a123a7fcc96a.png)
### 

- **STEP2**

![image](https://user-images.githubusercontent.com/58059527/183351523-9c0a22b2-ee80-4d07-a937-145491457ff9.png)
### 

我选择了第一种方法，我在`config.js`写入了
```
module.exports = {
  plugins: [
    registerComponentsPlugin({
      componentsDir: path.resolve(__dirname, './components'),
    }),
  ],
}
```

我的目录如下

```
├─ docs
│  ├─ .vuepress
│  ├─ │─ components
│  ├─ │ └─ A.vue
│  │  └─ config.js
│  └─ README.md
````

`README.md`中直接用`<A></A>`

:disappointed_relieved: 抱歉占用你的时间。
